pkg/http/api.go

```diff
@@ -231,6 +231,34 @@ func SetSecrets(host string, verifyTLS bool, apiKey string, project string, conf
 	return computed, Error{}
 }
 
+// SetSecretNote for specified project and config
+func SetSecretNote(host string, verifyTLS bool, apiKey string, project string, config string, secret string, note string) (models.SecretNote, Error) {
+	reqBody := map[string]interface{}{}
+	reqBody["secret"] = secret
+	reqBody["note"] = note
+	body, err := json.Marshal(reqBody)
+	if err != nil {
+		return models.SecretNote{}, Error{Err: err, Message: "Invalid secret note"}
+	}
+
+	var params []queryParam
+	params = append(params, queryParam{Key: "project", Value: project})
+	params = append(params, queryParam{Key: "config", Value: config})
+
+	statusCode, _, response, err := PostRequest(host, verifyTLS, apiKeyHeader(apiKey), "/v3/configs/config/secrets/note", params, body)
+	if err != nil {
+		return models.SecretNote{}, Error{Err: err, Message: "Unable to set secret note", Code: statusCode}
+	}
+
+	var result map[string]interface{}
+	err = json.Unmarshal(response, &result)
+	if err != nil {
+		return models.SecretNote{}, Error{Err: err, Message: "Unable to parse API response", Code: statusCode}
+	}
+
+	return models.SecretNote{Secret: result["secret"].(string), Note: result["note"].(string)}, Error{}
```

Comment:
Just want to confirm that if a secret does not have a note we send back:
```javascript
{
  secret: "SECRET",
  note: ""
}
```

Reply:
Confirmed, this endpoint always sends back a note of type `string`. When there's no note, it'll be an empty string.